oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/Fixer.java

```diff
@@ -0,0 +1,406 @@
+	  if (!started && !completed) {
+		started = true;
+		Runnable runner = new Runnable() {
+		  @Override
+		  public void run() {
+			try {
+			  BasicLTIPlacement placement = null;
+			  try {
+			  placement = BasicLTIPlacementManager.Factory.getInstance().loadByHandle(placementHandle);
+			  } catch (KeyNotFoundException ex) {
+				logMessage(0, "---------------------------------------------------");
+				logMessage(0, "An error occurred: The placement handle does not exist. Please introduce a valid placement handle code");
+				started = false;
+				completed = false;
+				errored = true;
+				throw ex;
+			  }
+			  logMessage(0, "Placement.getName: " + placement.getName());
+			  logMessage(0, "Placement url: " + placement.getUrl());
+			  logMessage(0, "Placement handle: " + placement.getHandle());
+			  logMessage(0, "Placement Id: " + placement.getId().toExternalString());
+
+			  mapToString(placement.getCustomParameters(), 0, "placement parameters");
+			  BbList courseList = new BbList();
+			  ;
+			  if (StringUtils.isNotEmpty(courseId) && courseDbLoader.doesCourseIdExist(courseId)) {
+				courseList.add(courseDbLoader.loadByCourseId(courseId));
+			  } else {
+				courseList = courseDbLoader.loadAllCourses();
+			  }
+			  int courseCount = courseList.size();
+			  for (int i = 0; i < courseCount; i++) {
+				percent = (int) (100.0 * i / courseCount);
+				Course course = (Course) courseList.get(i);
+
+				logMessage(0, "Looking at Course '" + course.getTitle() + "' ("
+				  + course.getId().toExternalString() + ")");
+
+
+				BbList courseTocs = courseTocDbLoader.loadByCourseId(course.getId());
+				for (int j = 0; j < courseTocs.size(); j++) {
+				  CourseToc courseToc = (CourseToc) courseTocs.get(j);
+
+				  recurseContent(contentDbLoader, courseToc,
+					getChildren(contentDbLoader, courseToc.getContentId()), placement.getId().toExternalString(), 3);
+				}
+			  }
+			  started = false;
+			  completed = true;
+			  errored = false;
+			} catch (Exception e) {
+			  started = false;
+			  completed = false;
+			  errored = true;
+			  logMessage(0, "---------------------------------------------------");
+			  logMessage(0, "An error occurred: " + e.getMessage());
+			  logMessage(0, "See Blackboard service logs for details");
+			  bbLogs.logError("An error occurred trying to fix EQUELLA content links", e);
+			}
+		  }
+		};
+		new Thread(runner, "Link Fixer Thread").start();
+	  }
+	}
+  }
+
+  protected void recurseContent(ContentDbLoader contentLoader, CourseToc courseToc, BbList contentList, String placementExternalId, int level)
+	throws Exception {
+	for (int j = 0; j < contentList.size(); j++) {
+
+	  Content content = (Content) contentList.get(j);
+	  lookedAt++;
+
+	  String handler = content.getContentHandler();
+
+	  boolean tleResource = handler.equals("resource/tle-resource") || handler.equals("resource/tle-myitem")
+		|| handler.equals("resource/tle-plan");
```

Comment:
are we reasonably confident these are the only old handles ?  Otherwise, might should make this a csv input on the Settings migration B2 page.

Reply:
This is what the old integration is creating. so yes